src/NeoSharp.Core/VM/InteropServiceExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using NeoSharp.VM;
+
+namespace NeoSharp.Core.VM
+{
+    public static class InteropServiceExtensions
+    {
+        public static void RegisterStackTransition(
+            this InteropService interopService,
+            string name,
+            Func<IStackAccessor, bool> handler)
+        {
+            bool BaseHandler(ExecutionEngineBase engine)
```

Comment:
I don't like methods inside other methods, the other code looks good. But do you reviewed if all SYSCALLS only use CurrentContext and `EvaluationStack`?

Reply:
I did. They don't 😄 Some of them will use an engine directly.